File: My Answer/E5.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class E5 {

    private static final Scanner scanner = new Scanner(System.in);

    private static String removeAllButLastDot(String s) {
        int last = s.lastIndexOf('.');
        if (last == -1) return s.replace(".", "");
        String before = s.substring(0, last).replace(".", "");
        String after = s.substring(last);
        if((s.length() - last) > 3) after = after.replace(".", "") + ".00";
        return before + after;
    }

    public static double extractDoubleValue(String msg){
        while(true){
            System.out.print(msg);
            String newPrice = scanner.nextLine();
            newPrice = newPrice.toUpperCase()
                                       .replace("RS","")
                                       .replace("R$","")
                                       .trim();
            double numPrice = 0.0;
            try {
                numPrice = Double.parseDouble(newPrice);
                return numPrice;
                
            } catch (NumberFormatException e) {
                newPrice = newPrice.replace(",", ".");
                newPrice = removeAllButLastDot(newPrice);
                try {
                    numPrice = Double.parseDouble(newPrice);
                    return numPrice;
                } catch (NumberFormatException e3) {
                    System.out.println("❌ Formato inválido! Digite um valor numérico válido.");
                }
            }
        }
    }

    public static void Q1(){
        double valor = extractDoubleValue("Digite o valor que deseja investir: ");
        if(valor < 1000.00){
            System.out.println("Perfil iniciante: Sugerimos Tesouro Direto");
        }else if(valor < 5000.00){
            System.out.println("Perfil moderado: Sugerimos Fundos Imobiliários");
        }else{
            System.out.println("Perfil arrojado: Sugerimos Ações");
        }
    }

    public static void Q2(){
        List<String> admins = new ArrayList<>(Arrays.asList("ana@empresa.com", "guilherme@empresa.com", "felipe@empresa.com"));
        System.out.print("digite seu email: ");
        String email = scanner.nextLine();
        email = email.replaceAll("\\s", "").toLowerCase();
        int admin_index = admins.indexOf(email);
        System.out.println((admin_index != -1) ? "Acesso liberado! Bem-vindo ao painel de controle.":"Acesso negado. Você não tem permissões de administrador.");

    }

    public static void Q3(){
        double valor_compra = extractDoubleValue("Digite o valor total da compra: ");
        double desconto = 0.00;
        if(valor_compra >= 500.00){
            System.out.println("15% de desconto:");
            desconto = valor_compra*0.15;
        }else if(valor_compra >= 200.00){
            System.out.println("10% de desconto:");
            desconto = valor_compra*0.1;
        }else{
            System.out.println("Sem desconto:");
        }
        System.out.printf("O valor total para pagar é de R$%.2f com desconto de R$%.2f.\n", (valor_compra - desconto), desconto);
    }

    public static void Q4(){
        double minhas_vendas = extractDoubleValue("Digite o valor total das suas vendas: ");
        double minha_meta = extractDoubleValue("Digite sua meta fiscal: ");
        double loja_vendas = extractDoubleValue("Digite o valor total das vendas da sua loja: ");
        double loja_meta = extractDoubleValue("Digite a meta fiscal da loja: ");
        double bonus = 0.0;
        if((minhas_vendas/minha_meta >= 1.0) && (loja_vendas/loja_meta >= 1)){
            bonus = minhas_vendas*0.2;
        }
        System.out.printf("Seu bônus este mês é de: R$%.2f\n", bonus);
    }

    public static void Q5(){
        System.out.print("Digite o assunto a ser tratado: ");
        String assunto = scanner.nextLine();
        assunto = assunto.toLowerCase();
        if(assunto.contains("pagamento") || assunto.contains("boleto") || assunto.contains("dívida") || assunto.contains("pagar")){
            System.out.println("Encaminhado para o Financeiro.");
        }else if(assunto.contains("entrega") || assunto.contains("atraso")){
            System.out.println("Encaminhado para a Logística");
        }else{
            System.out.println("Encaminhado para o Suporte Geral");
        }
    }
    public static void main(String[] args) {
        //System.out.println("Validação de Investimento (Setor Financeiro)");
        //Q1();
        //System.out.println("Controle de Acesso ao Sistema (Setor de Segurança)");
        //Q2();
        //System.out.println("Cálculo de Desconto Progressivo (Setor de Vendas)");
        //Q3();
        //System.out.println("Análise de Metas Combinadas (Setor Comercial)");
        //Q4();
        System.out.println("Sistema de Triagem de E-mails (Setor de Customer Experience)");
        Q5();
    }
}

File: My Answer/E5.cpp
#include <iostream>
#include <string>
#include <vector>
#include <stdio.h>
#include <stdlib.h>
#include <omp.h>
#include <iomanip>
#include <algorithm>
#include <cctype>
#include <locale>
#include <regex>
#include <cmath>
#include <stdexcept>
#include "utils.h"

using namespace std;

void Q1(){
    cout<<"Digite o valor de seu investimento: ";
    float valor_investir = extractBillingValue();
    if(valor_investir < 1000.0){
        cout << "Perfil Iniciante: sugerimos Tesouro Direto.";
    }else if(valor_investir < 5000.0){
        cout << "Perfil Moderado: sugerimos Fundos Imobiliários.";
    }else{
        cout << "Perfil Arrojado: sugerimos Ações.";
    }
    cout << endl;
}

void Q2(){
    vector<string> admins = {"ana@empresa.com", "guilherme@empresa.com", "felipe@empresa.com"};
    bool flag = false;
    cout << "Digite seu email: ";
    string my_email;
    getline(cin, my_email);
    replaceAll(my_email, " ", "");
    transform(my_email.begin(), my_email.end(), my_email.begin(), ::tolower);
    cout<<"Seu email: "<<my_email<<endl;
    for(int i = 0; i < admins.size(); i++){
        if(my_email == admins[i]){
            flag = true;
            break;
        }
    }
    cout << (flag ? "Acesso Liberado! Bem vindo ao Painel de Controle!":"Acesso Negado. Você não tem permissões de admnistrador.")<<endl;
}

void Q3(){
    cout<<"Digite o valor total de suas compras: ";
    float valor = extractBillingValue();
    cout<<"Valor total de compras: R$"<< fixed << setprecision(2) << valor << endl;
    float desconto = 0.0;
    if(valor >= 500){
        cout << "Desconto de 15%." << endl;
        desconto = (valor*0.15);
    }else if(valor >= 200){
        cout << "Desconto de 10%." << endl;
        desconto = (valor*0.10);
    }else{
        cout << "Sem desconto." << endl;
    }
    float total_a_pagar = (valor - desconto);
    cout<<"Total a Pagar: R$" << fixed << setprecision(2) << total_a_pagar << endl;
    cout<<"Valor do Desconto: R$" << fixed << setprecision(2) << desconto << endl;
}

void Q4(){
    float vendas = 0.0;
    float temp = 0.0;
    while(true){
        cout << "digite o valor de suas vendas ou 0 para saír: ";
        temp = extractBillingValue();
        if(temp == 0.0)break;
        vendas += temp;
    }
    float meta = 0;
    cout << "digite sua meta de vendas: ";
    meta = extractBillingValue();
    cout << "Seu bônus esse mês é de: R$" << fixed << setprecision(2) << ((vendas >= meta) ? (vendas*0.2):0.0) << endl;
}

void Q5(){
    string user_request;
    cout << "Bem vindo ao Sistema de Atendimento ao Consumidor. O que deseja?\n";
    getline(cin >> ws, user_request);
    transform(user_request.begin(), user_request.end(), user_request.begin(), ::tolower);
    if (user_request.find("pagamento") != string::npos || user_request.find("boleto") != string::npos) {
        cout << "Resultado: Encaminhado para o Financeiro." << endl;
    } 
    else if (user_request.find("entrega") != string::npos || user_request.find("atraso") != string::npos) {
        cout << "Resultado: Encaminhado para a Logística." << endl;
    } 
    else {
        // Esta é a terceira condição que apareceu na segunda página da imagem:
        cout << "Resultado: Encaminhado para o Suporte Geral." << endl;
    }
}

int main(){
    //Q1();
    //Q2();
    //Q3();
    //Q4();
    Q5();
    return 0;
}